### DISCUSSION TITLE
Slooooowness

### DISCUSSION THREAD

**Discussion by hpatner** [mdelobelle/metadatamenu#530] [Q&A]
I just started to get into Metadatamenu. It is totally changing my work flow and is amazing. The issue, I'm having is that my system now seems to get very significant lags.  I wondering if there are any ideas on how to speed it up.
I wonder if it has something to do with the queries I set up for selecting fields.  Any pointers would be amazing. 

Here is an example.
![image](https://github.com/mdelobelle/metadatamenu/assets/5373835/cc18c1ac-8865-438c-90cc-0e37ae0feee5)

**Comment by mdelobelle** [comment]
Those queries look fine. 
Even if they return loads of results, they are executed when the corresponding field modal is triggered: so this won't affect the overall performance.

Things that take time (and cpu):
Indexing the "managed fields" for each single note. Done for all files at start and for any file each time this file is changed.
During this process the whole file is parsed.
If you have long files it will take a lot of time each time obsidian will reindex your file (basically every 500 ms or so when you type text with the keyboard)

If your metadata only live in frontmatter (and not inline with the dataview key::value syntax) you can set `scope : frontmatter only` in the settings. The plugin will limit the parsing to the frontmatter and this could accelerate the process by a lot

**Comment by hpatner** [comment]
mdelabelle - thank you so much. I already had it set to search frontmatter only but was super slow.
I wound up turning "auto calculation" off and it is much zippier.
Wondering if auto calculation is off, do I need to close obsidian and reopen to have it recalculate the index? Not totally clear yet on what I'm losing from toggling this off.

![image](https://github.com/mdelobelle/metadatamenu/assets/5373835/0e906466-a4d4-431a-9a2a-ec3c683a88eb)

**Comment by mdelobelle** [comment]
Great. To refresh on demand, just use the "update all lookups and formulas" command
![image](https://github.com/mdelobelle/metadatamenu/assets/1233240/abff4b18-0fb1-4eb3-b2bb-ff6c13f31de5)

EDIT: turning-off autocalculation will prevent Metadata Menu from recalculating Formula and Lookups after each file cache change by Obsidian.
I didn't k'ow you where using Formula or Lookups by yeah, if it's the case in an extensive manner, toggling off Auto-calcultion is the way to go.

When doing so: the files that have a formula or lookup in it will have their icon turn "orange" to show that there "may be" updates to calculate. Then you can use either:
- the update all formulas and lookups command
- click on the index update icon in the status bar (the folder icon with a red pencil) to update the active file

**Comment by mdelobelle** [comment]
Moving it to the discussion section in case someone experience the same situation
